# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundParentImpl.cpp
# Commit: 296b2617fceb
# Full Hash: 296b2617fceb82bd87efffda2aabe38af402f7af
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-05-06 16:28:18
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Make Factory derive from SafeRefCounted. r=dom-workers-and-storage-reviewers,janv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69398
# ==============================================================================

diff -r 74d93e32c149 -r 296b2617fceb ipc/glue/BackgroundParentImpl.cpp
--- a/ipc/glue/BackgroundParentImpl.cpp	Wed May 06 11:35:20 2020 +0000
+++ b/ipc/glue/BackgroundParentImpl.cpp	Wed May 06 12:20:14 2020 +0000
@@ -167,7 +167,8 @@
 }
 
 auto BackgroundParentImpl::AllocPBackgroundIDBFactoryParent(
-    const LoggingInfo& aLoggingInfo) -> PBackgroundIDBFactoryParent* {
+    const LoggingInfo& aLoggingInfo)
+    -> already_AddRefed<PBackgroundIDBFactoryParent> {
   using mozilla::dom::indexedDB::AllocPBackgroundIDBFactoryParent;
 
   AssertIsInMainOrSocketProcess();
@@ -191,17 +192,6 @@
   return IPC_OK();
 }
 
-bool BackgroundParentImpl::DeallocPBackgroundIDBFactoryParent(
-    PBackgroundIDBFactoryParent* aActor) {
-  using mozilla::dom::indexedDB::DeallocPBackgroundIDBFactoryParent;
-
-  AssertIsInMainOrSocketProcess();
-  AssertIsOnBackgroundThread();
-  MOZ_ASSERT(aActor);
-
-  return DeallocPBackgroundIDBFactoryParent(aActor);
-}
-
 auto BackgroundParentImpl::AllocPBackgroundIndexedDBUtilsParent()
     -> PBackgroundIndexedDBUtilsParent* {
   AssertIsInMainOrSocketProcess();